tests/qubit_codes/test_surface_code.py

```diff
@@ -272,6 +274,29 @@ def test_periodic_boundaries(self, d):
 
         assert not set(RHG_lattice.edges) - set(RHG_graph(d, "periodic").edges)
 
+    @pytest.mark.parametrize("boundary_type", ["periodic_primal", "periodic_dual"])
+    def test_toric_boundaries(self, d, boundary_type):
+        """Test whether periodic boundary conditions on x and y direction
+        produce a lattice with the expected neighbors."""
+        RHG_lattice = RHG_graph(d, boundary_type)
+
+        for plane in ("x", "y"):
+            lower_boundary = RHG_lattice.slice_coords(plane, 0)
+
+            for point in lower_boundary:
+                # get neighbors tagged as periodic
+                neighbors = RHG_lattice[point]
+                periodic_neighbors = [
+                    coords for coords, props in neighbors.items() if "periodic" in props
+                ]
+
+                # check there is an edge from neighbor in low boundary to high boundary
+                # for all periodic neighbors
+                assert all(
+                    RHG_lattice.has_edge(point, periodic_point)
+                    for periodic_point in periodic_neighbors
+                )
+
```

Comment:
Let us actually rethink this test. A few things:
- We may as well also test the `periodic` setting.
- Currently the test is trivial: it checks whether `point` and `periodic_point` have an edge, but they must have an edge by definition because they are neighbours of each other.
- I think we can recreate the logic from lines 225-234 of surface_code.py : do the nodes on the low boundary and the high boundary that we expect to have an edge between them, along with the ``periodic`` tag, actually have it. (Alternatively, we can do a counting argument: are there the expected number of periodic edges on each boundary. This will require a little bit of work to find the pattern).

Reply:
> I think we can recreate the logic from lines 225-234 of surface_code.py : do the nodes on the low boundary and the high boundary that we expect to have an edge between them, along with the periodic tag, actually have it. (Alternatively, we can do a counting argument: are there the expected number of periodic edges on each boundary. This will require a little bit of work to find the pattern).

Does it actually make sense to test using the same logic? I'd be more in favor of using an unrelated logic, i.e., _are there the expected number of periodic edges on each boundary_; however I will need some help to figure the pattern out.

Reply:
Done! 💪